src/ios/PurchasesPlugin.swift

```diff
@@ -1,10 +1,397 @@
+            self.commandDelegate.send(result, callbackId: command.callbackId)
+        }
+    }
+
+    @objc(purchaseProduct:) func purchaseProduct(command: CDVInvokedUrlCommand) {
+        guard let productIdentifier = command.arguments[0] as? String else {
+            self.sendBadParameterFor(command: command, parameterNamed: "productIdentifier", expectedType: NSString.self)
+            return
+        }
+
+        RCCommonFunctionality.purchaseProduct(productIdentifier,
+                                              signedDiscountTimestamp: nil,
+                                              completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+    @objc(purchasePackage:) func purchasePackage(command: CDVInvokedUrlCommand) {
+        guard let packageIdentifier = command.arguments[0] as? String,
+        let offeringIdentifier = command.arguments[1] as? String else {
+            self.sendBadParametersFor(command: command,
+                                      parametersNamed: ["packageIdentifier", "offeringIdentifier"],
+                                      expectedTypes: [NSString.self, NSString.self])
+            return
+        }
+
+        RCCommonFunctionality.purchasePackage(packageIdentifier,
+                                              offering: offeringIdentifier,
+                                              signedDiscountTimestamp: nil,
+                                              completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+    @objc(restoreTransactions:) func restoreTransactions(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.restoreTransactions(completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+    @objc(getAppUserID:) func getAppUserID(command: CDVInvokedUrlCommand) {
+        let result = CDVPluginResult(status: .ok, messageAs: RCCommonFunctionality.appUserID())
+        self.commandDelegate.send(result, callbackId: command.callbackId)
+    }
+
+    @objc(logIn:) func logIn(command: CDVInvokedUrlCommand) {
+        guard let appUserID = command.arguments[0] as? String else {
+            self.sendBadParameterFor(command: command, parameterNamed: "appUserID", expectedType: NSString.self)
+            return
+        }
+
+        RCCommonFunctionality.logIn(withAppUserID: appUserID,
+                                    completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+    @objc(logOut:) func logOut(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.logOut(completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+
+    @objc(setupShouldPurchasePromoProductCallback:) func setupShouldPurchasePromoProductCallback(command: CDVInvokedUrlCommand) {
+        self.shouldPurchasePromoProductCallbackID = command.callbackId;
+        // TODO: This next line is needed, right? It wasn't in the old version.
+        self.sendOKFor(command: command)
```

Comment:
I'm not sure, but I think we _don't_ want to send ok for that command yet - what this is doing is to store a callback for later use if I'm understanding correctly. so sending ok for it might actually be killing the callback

Reply:
There we go, ok.